http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala

```diff
@@ -400,11 +471,157 @@ class GKEInterpreter[F[_]: Parallel: ContextShift: Timer](
+
+      helmAuthContext <- getHelmAuthContext(googleCluster, dbCluster.id, namespaceName)
+
+      releaseName = ReleaseName(s"${appName.value}-${config.galaxyAppConfig.releaseNameSuffix.value}")
+      chartValues = buildGalaxyChartOverrideValuesString(appName,
+                                                         releaseName,
+                                                         dbCluster,
+                                                         nodepoolName,
+                                                         userEmail,
+                                                         customEnvironmentVariables,
+                                                         kubernetesServiceAccount)
+
+      _ <- logger.info(
+        s"Chart override values are: ${chartValues} | trace id: ${ctx.traceId}"
+      )
+
+      // Invoke helm
+      _ <- helmClient
+        .installChart(
+          org.broadinstitute.dsp.Release(releaseName.value),
+          org.broadinstitute.dsp.Chart(config.galaxyAppConfig.chart.value),
+          org.broadinstitute.dsp.Values(chartValues)
+        )
+        .run(helmAuthContext)
+
+      // Poll galaxy until it starts up
+      // TODO potentially add other status checks for pod readiness, beyond just HTTP polling the galaxy-web service
+      isDone <- (
+        Stream.sleep_(config.monitorConfig.createApp.interval) ++
+          Stream.eval(
+            galaxyDAO.isProxyAvailable(dbCluster.googleProject, appName)
+          )
+      ).repeatN(config.monitorConfig.createApp.maxAttempts)
+        .takeThrough(_ == false)
+        .compile
+        .lastOrError
+
+      _ <- if (!isDone) {
+        val msg =
+          s"Galaxy installation has failed or timed out for app ${appName.value} in cluster ${dbCluster.getGkeClusterId.toString} | trace id: ${ctx.traceId}"
+        logger.error(msg) >>
+          F.raiseError[Unit](AppCreationException(msg))
+      } else F.unit
+
+    } yield ()
+
+  private[util] def uninstallGalaxy(dbCluster: KubernetesCluster,
+                                    appName: AppName,
+                                    namespaceName: NamespaceName,
+                                    googleCluster: Cluster)(
+    implicit ev: ApplicativeAsk[F, AppContext]
+  ): F[Unit] =
+    for {
+      ctx <- ev.ask
+
+      _ <- logger.info(
+        s"Uninstalling helm chart ${config.galaxyAppConfig.chart.value} for app ${appName.value} in cluster ${dbCluster.getGkeClusterId.toString} | trace id: ${ctx.traceId}"
+      )
+
+      helmAuthContext <- getHelmAuthContext(googleCluster, dbCluster.id, namespaceName)
+
+      releaseName = ReleaseName(s"${appName.value}-${config.galaxyAppConfig.releaseNameSuffix.value}")
+
+      // Invoke helm
+      _ <- helmClient
+        .uninstall(org.broadinstitute.dsp.Release(releaseName.value), config.galaxyAppConfig.uninstallKeepHistory)
+        .run(helmAuthContext)
+
+      isDone <- (Stream.sleep_(config.monitorConfig.deleteApp.interval) ++
```

Comment:
same. you should be able to use `streamFUntilDone` and provide a `DoneCheckable[List[Pod]]` instance (hand wavy signature)

Reply:
Forgot about `streamFUntilDone`! Thanks